Packages/Gist/Sources/Gist/GistLists/GistListsViewModel.swift

```diff
@@ -26,32 +30,49 @@ public final class GistListsViewModel: ObservableObject {
 
     func fetchGists(listsMode: GistListsMode) async {
         do {
-            let gists: [Gist]
+            isLoadingMoreGists = true
             switch listsMode {
             case .currentUserGists:
-                gists = try await client.gists()
+                let gistsResponse = try await client.gists(pageSize: Constants.pagingSize, cursor: pagingCursor)
+                pagingCursor = gistsResponse.cursor
+                hasMoreGists = gistsResponse.hasNextPage
+                gists.append(contentsOf: gistsResponse.gists)
             case .currentUserStarredGists:
                 gists = try await client.starredGists()
             case let .userGists(userName):
-                gists = try await client.gists(fromUserName: userName)
+                let gistsResponse = try await client.gists(fromUserName: userName, pageSize: Constants.pagingSize, cursor: pagingCursor)
+                pagingCursor = gistsResponse.cursor
+                hasMoreGists = gistsResponse.hasNextPage
+                gists.append(contentsOf: gistsResponse.gists)
             }
-            self.gists = gists
-            contentState = .content(gists: gists)
+            isLoadingMoreGists = false
+            contentState = .content
         } catch {
             contentState = .error(error: error.localizedDescription)
         }
     }
 
+    func fetchMoreGistsIfNeeded(lastGistId: String, listsMode: GistListsMode) async {
+        guard listsMode != .currentUserStarredGists else {
+            return
+        }
+        guard hasMoreGists, !isLoadingMoreGists, lastGistId == gists.last?.id ?? "" else {
+            return
+        }
+        await fetchGists(listsMode: listsMode)
+    }
+
     func insert(_ gist: Gist) {
         gists.insert(gist, at: 0)
-        contentState = .content(gists: gists)
     }
 
-    func search() {
+    func search(listMode: GistListsMode) {
         if searchText.isEmpty {
-            contentState = .content(gists: self.gists)
+            Task {
+                await fetchGists(listsMode: listMode)
```

Comment:
Suggestion: Use current gists of `viewModel` to avoid API called once the search text is empty